cordformation/src/main/kotlin/net/corda/plugins/Dockerform.kt

```diff
@@ -177,4 +182,24 @@ open class Dockerform @Inject constructor(objects: ObjectFactory) : Baseform(obj
 
         Files.write(dockerComposePath, dockerComposeContent.toByteArray())
     }
+
+    private fun encodeTypesafeURL(url: String): String {
+        val matcher = Pattern.compile("\\$\\{([^}]+)\\}").matcher(url);
```

Comment:
The `Pattern` only needs to be compiled once, and then you can use it to create as many `Matcher` objects as you like, e.g.
```kotlin
val matcher = PRECOMPILED_PATTERN.matcher(url)
```
IntelliJ also informs me that you don't need to escape the `}` character at the end, and that there's an unnecessary semi-colon there :wink:.